File: regular_assignments/day_024/fibonacci_2/solution.rb
class Solution
  def call
    n = 2
    puts "n: #{n}, val: #{findAthFibonacci(n)}"
    n = 9
    puts "n: #{n}, val: #{findAthFibonacci(n)}"
  end

  def findAthFibonacci(n)
    fibonacci(n)
  end

  def fibonacci(n)
    if n == 0
      return 0
    elsif n == 1
      return 1
    else
      return fibonacci(n-1) + fibonacci(n-2)
    end
  end
end

File: regular_assignments/day_024/fibonacci_2/solution.go
package fibonacci_2

import "fmt"

func Call() {
	fmt.Printf("n: %v, val: %v\n", 2, findAthFibonacci(2))
	fmt.Printf("n: %v, val: %v\n", 9, findAthFibonacci(9))
}

func findAthFibonacci(n int) int {
	return fibonacci(n)
}

func fibonacci(n int) int {
	if n == 0 {
		return 0
	} else if n == 1 {
		return 1
	} else {
		return fibonacci(n-1) + fibonacci(n-2)
	}
}
